apps/omg_db/lib/leveldb_server.ex

```diff
@@ -53,92 +52,92 @@ defmodule OMG.DB.LevelDBServer do
       |> Kernel.<>(".Recorder")
       |> String.to_atom()
 
-    {:ok, _} = Recorder.start_link(%Recorder{name: name, parent: self()})
+    {:ok, _recorder_pid} = Recorder.start_link(%Recorder{name: name, parent: self()})
 
     with {:ok, db_ref} <- Exleveldb.open(db_path, create_if_missing: false) do
-      {:ok, %__MODULE__{db_ref: db_ref}}
+      {:ok, %__MODULE__{name: name, db_ref: db_ref}}
     else
       error ->
         _ = Logger.error("It seems that Child chain database is not initialized. Check README.md")
         error
     end
   end
 
-  def handle_call({:multi_update, db_updates}, _from, %__MODULE__{db_ref: db_ref} = state) do
+  def handle_call({:multi_update, db_updates}, _from, %__MODULE__{db_ref: db_ref, name: name} = state) do
     result =
       db_updates
       |> LevelDBCore.parse_multi_updates()
-      |> write(db_ref)
+      |> write(db_ref, name)
 
     {:reply, result, state}
   end
 
-  def handle_call({:blocks, blocks_to_fetch}, _from, %__MODULE__{db_ref: db_ref} = state) do
+  def handle_call({:blocks, blocks_to_fetch}, _from, %__MODULE__{db_ref: db_ref, name: name} = state) do
     result =
       blocks_to_fetch
       |> Enum.map(fn block -> LevelDBCore.key(:block, block) end)
-      |> Enum.map(fn key -> get(key, db_ref) end)
+      |> Enum.map(fn key -> get(key, db_ref, name) end)
       |> LevelDBCore.decode_values(:block)
 
     {:reply, result, state}
   end
 
-  def handle_call(:utxos, _from, %__MODULE__{db_ref: db_ref} = state) do
-    result = get_all_by_type(:utxo, db_ref)
+  def handle_call(:utxos, _from, %__MODULE__{db_ref: db_ref, name: name} = state) do
+    result = get_all_by_type(:utxo, db_ref, name)
     {:reply, result, state}
   end
 
-  def handle_call(:exit_infos, _from, %__MODULE__{db_ref: db_ref} = state) do
-    result = get_all_by_type(:exit_info, db_ref)
+  def handle_call(:exit_infos, _from, %__MODULE__{db_ref: db_ref, name: name} = state) do
+    result = get_all_by_type(:exit_info, db_ref, name)
     {:reply, result, state}
   end
 
-  def handle_call({:block_hashes, block_numbers_to_fetch}, _from, %__MODULE__{db_ref: db_ref} = state) do
+  def handle_call({:block_hashes, block_numbers_to_fetch}, _from, %__MODULE__{db_ref: db_ref, name: name} = state) do
     result =
       block_numbers_to_fetch
       |> Enum.map(fn block_number -> LevelDBCore.key(:block_hash, block_number) end)
-      |> Enum.map(fn key -> get(key, db_ref) end)
+      |> Enum.map(fn key -> get(key, db_ref, name) end)
       |> LevelDBCore.decode_values(:block_hash)
 
     {:reply, result, state}
   end
 
-  def handle_call(:in_flight_exits_info, _from, %__MODULE__{db_ref: db_ref} = state) do
-    result = get_all_by_type(:in_flight_exit_info, db_ref)
+  def handle_call(:in_flight_exits_info, _from, %__MODULE__{db_ref: db_ref, name: name} = state) do
+    result = get_all_by_type(:in_flight_exit_info, db_ref, name)
```

Comment:
since we have two fields of `state` used now, maybe we can consider passing `state` around and pattern matching the details DRYly in functions like `get_all_by_type`?